lib/logstash/filters/jdbc/db_object.rb

```diff
@@ -0,0 +1,148 @@
+require_relative "validatable"
+require_relative "column"
+
+module LogStash module Filters module Jdbc
+
+  TEMP_TABLE_PREFIX = "temp_".freeze
+  DB_OBJECT_TYPES = ["table".freeze, "index".freeze].freeze
+
+  class DbObject < Validatable
+    #   {type => "table", name => "servers", columns => [["ip", "text"], ["name", "text"], ["location", "text"]]},
+    #   {type => "index", name => "servers_idx", table => "servers", columns => ["ip"]}
+
+    attr_reader :type, :name, :table, :columns, :builder
+
+    def build(db)
+      return unless valid?
+      if index?
+        IndexBuilder.new(@name, @columns, @table).build(db)
+      else
+        TableBuilder.new(@name, @columns).build(db)
+      end
+    end
+
+    class TableBuilder
+      attr_reader :name, :columns
+      def initialize(name, columns)
+        @name, @columns = name, columns
+      end
+      def build(db)
+        schema_gen = db.create_table_generator()
+        @columns.each {|col| schema_gen.column(col.name, col.datatype)}
+        options = {:generator => schema_gen}
+        db.create_table(@name, options)
+      end
+    end
+
+    class IndexBuilder
+      attr_reader :name, :columns, :table
+      def initialize(name, columns, table)
+        @name, @columns, @table = name, columns, table
+      end
+      def build(db)
+        db.add_index(@table, @columns, :name => @name)
```

Comment:
This might raise and error? If the user doesn't have the privileges to create the index.

Reply:
I might sound alarming, but I've been bite in the past with ultra strict DBA :)